pkg/reconciler/buildrun/resources/results.go

```diff
@@ -0,0 +1,98 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package resources
+
+import (
+	"fmt"
+
+	build "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+
+	pipeline "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+)
+
+type sourceResult struct {
+	defaultSource,
+	bundleSource build.SourceResult
+}
+
+const (
+	defaultSourceName       = "default"
+	commitSHAResult         = "commit-sha"
+	commitAuthorResult      = "commit-author"
+	bundleImageDigestResult = "bundle-image-digest"
+	imageDigestResult       = "image-digest"
+	imageSizeResult         = "image-size"
+)
+
+// UpdateBuildRunUsingTaskResults surface the task results
+// to the buildrun
+func UpdateBuildRunUsingTaskResults(
+	buildRun *build.BuildRun,
+	lastTaskRun *pipeline.TaskRun,
+) {
+	var sources sourceResult
+
+	// Initializing output result
+	buildRun.Status.Output = &build.Output{}
+
+	for _, result := range lastTaskRun.Status.TaskRunResults {
+		updateBuildRunStatus(buildRun, result, &sources)
+	}
+
+	// Appending the source result only if the defined source
+	// from build spec emitting the results
+	if sources.defaultSource.Name != "" {
+		buildRun.Status.Sources = append(buildRun.Status.Sources, sources.defaultSource)
+	}
+
+	if sources.bundleSource.Name != "" {
+		buildRun.Status.Sources = append(buildRun.Status.Sources, sources.bundleSource)
+	}
+}
+
+func updateBuildRunStatus(
```

Comment:
The handling of the output image results is fine here. I would move the source-related things into https://github.com/shipwright-io/build/blob/main/pkg/reconciler/buildrun/resources/sources.go and the type-specific details (git / bundle) into their specific files like https://github.com/shipwright-io/build/blob/main/pkg/reconciler/buildrun/resources/sources/git.go. Then, the creation of the result and the parsing of it is in the same file.